### PR TITLE
feat(operation): general (non-symmetric) eigenvector generator + eigen API (#203)

### LINKED ISSUE #203: feat(operation): general (non-symmetric) eigenvector generator + unified eigen API
## Problem

`mtl::eigenvalue(A)` (`include/mtl/operation/eigenvalue.hpp:18`) computes eigenvalues of a general (non-symmetric) dense matrix via Hessenberg reduction + Francis QR, but returns **eigenvalues only**. There is no way to obtain eigenvectors for the general case. Only the symmetric path (`eigen_symmetric`) produces eigenvectors.

## Goal

Add a general-matrix **eigenvector generator** and a unified eigen API that returns eigenvalues and (optionally) eigenvectors.

## Scope

- Compute right eigenvectors for a general real dense matrix. Options to consider:
  - Accumulate the QR Schur vectors during Francis iteration, then back-substitute on the (quasi-)triangular Schur form to recover eigenvectors (LAPACK `trevc`-style).
  - Handle real + complex-conjugate eigenpairs (2×2 Schur blocks) and return complex eigenvectors.
- Public API, e.g. `mtl::eigen(A)` returning a struct `EigenResult{ dense_vector<complex<T>> eigenvalues; dense2D<complex<T>> eigenvectors; }`, structured-bindable, with column *k* the eigenvector for `eigenvalues(k)` — mirroring `eigen_symmetric`'s shape.
- Optionally left eigenvectors as a follow-up.

## Acceptance criteria

- `A·v_k = λ_k·v_k` residual within tolerance for real and complex-conjugate pairs.
- Reconstruction / spectral checks on generator matrices (Wilkinson, Clement, Frank, Forsythe, random).
- Unit tests in `tests/unit/operation/` (extend `test_eigenvalue.cpp` or add `test_eigen.cpp`).
- Docs page under `docs/` and entry wired into the docs pipeline.

Part of the eigen-enhancements epic.

### PR DESCRIPTION
**Pull request by Ravenwater**
Closes #203. Part of epic #202.

## What

Adds `mtl::eigen(A)` — eigenvalues **and right eigenvectors** for a general
(non-symmetric) square matrix, filling the gap where `eigenvalue(A)` returned
values only.

```cpp
auto [eigs, V] = mtl::eigen(A);   // column k of V is the eigenvector for eigs(k)
// A * V(:,k) = eigs(k) * V(:,k)
```

Returns a structured-bindable `{ eigenvalues, eigenvectors }` (both `complex`),
mirroring `eigen_symmetric`. Eigenvectors are unit 2-norm with a canonical
phase (largest entry real-positive) for deterministic output.

## How

- Eigenvalues from the existing, tested general QR path (`eigenvalue`).
- Each eigenvector via **inverse iteration** on `A - lambda_k*I`, using a small
  partial-pivot complex LU with a pivot floor so the (near-singular) system
  stays solvable — the regime inverse iteration relies on. Real and
  complex-conjugate pairs are handled uniformly in complex arithmetic.
- Chosen over Schur+`trevc` because it reuses the validated eigenvalue path
  rather than rewriting the Francis iteration to accumulate Schur vectors
  (far less error-prone for a reference implementation). The LAPACK `geev`
  path (#204) will supersede it for float/double.

## Tests (`tests/unit/operation/test_eigen_general.cpp`, GCC + Clang)

Known real eigenvalues, a genuine complex-conjugate pair, diagonal, 1x1, 0x0,
Frank (n=6, non-symmetric real spectrum), a mixed real+complex-conjugate 4x4,
and consistency with `eigenvalue()`. Acceptance metric is the eigen residual
`max_i |(A v_k - lambda_k v_k)_i|` (< 1e-8..1e-10) plus unit-norm checks.

## Docs

`docs/algorithms/eigenvalues.md` — "which routine do I call?" table + the
general eigen API, wired into `FILE_MAP`. Seeds the fuller guide in #207.

## Known limitation (discovered, pre-existing)

`eigenvalue()`'s single-shift real QR **stalls on strongly non-normal matrices**
whose complex eigenvalues need a double-shift (Francis) step — e.g. the Forsythe
companion matrix, where it returns `3,3,3,3,3` (all shift value) instead of the
true radius-0.87 circle. The existing Forsythe test only checked trace-sum and a
loose distance bound, so this was masked. `eigen`'s accuracy inherits this bound;
the doc-comment and docs page call it out, and it's the province of #204
(LAPACK `geev`) / a double-shift QR fix — **out of scope** for the eigenvector
generator itself. Happy to file a dedicated issue for the `eigenvalue()`
double-shift gap if you'd like.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added `eigen` functionality for general square matrices, returning eigenvalues and normalized right eigenvectors.
  * Eigenvectors use a consistent, deterministic sign/phase convention.
* **Documentation**
  * Added comprehensive guidance and examples for general and symmetric eigenvalue routines, including backend and accuracy details.
* **Tests**
  * Added coverage for real and complex spectra, diagonal and block matrices, edge cases, eigenvector normalization, and residual accuracy.

### CODE REVIEW

**Comment by Ravenwater** [comment]
@coderabbitai review

File: include/mtl/operation/eigenvalue.hpp [hunk 1]
```diff
@@ -162,4 +221,123 @@ auto eigenvalue(const M& A, typename M::value_type tol = 1e-10,
...
+        std::vector<std::size_t> piv(nn);
+        detail::eig_lu_factor(Mc, piv, nn, pivot_floor);
+
+        // Start vector: entries varied by index so that repeated/clustered
+        // eigenvalues are seeded with non-parallel starts.
+        std::vector<complex_type> x(nn);
+        for (size_type i = 0; i < n; ++i)
+            x[static_cast<std::size_t>(i)] =
+                complex_type(value_type(1) + value_type(i) / value_type(n));
+
```
[the remark this answers is not in the document]
Comment (Ravenwater) [reply include/mtl/operation/eigenvalue.hpp]: Fixed in d28d70b. Confirmed the defect: inverse iteration used an identical seed and factorization for every equal eigenvalue, so `eigen(I)` returned duplicate columns (residual was still 0 since any vector is an eigenvector of I, which is why it slipped the original tests).

Resolution: deflated inverse iteration — a `k`-dependent seed plus modified Gram-Schmidt against previously computed eigenvectors sharing the same eigenvalue, matched within a tiny `cluster_tol = sqrt(eps)*(||A||_F + |lambda|)`. That tolerance is deliberately tiny so genuinely distinct eigenvalues — importantly, complex-conjugate partners (whose eigenvectors are conjugate, not orthogonal) — are never merged/orthogonalized. Successive iterates stay in the orthogonal complement of what is already found, giving an independent orthonormal basis of the eigenspace (orthonormal combinations within one eigenspace are still eigenvectors).

Added regression tests: identity (multiplicity 4 → orthonormal basis via Gram check) and partial multiplicity `diag(2,2,5)`. 10 cases / 57 assertions pass on GCC and Clang.
